File: typescript/patterns-use-cases/src/webhookcallbacks/stubs.ts
import * as restate from "@restatedev/restate-sdk";

export type StripeEvent = {
    id: string; // event id
    type: string;
    created: number;
    data: {
        object: {
            id: string; // invoice id
            customer: string;
        };
    };
}

// Have a look at the scheduling tasks example (../schedulingtasks/payment_reminders.ts)
// to see a full implementation of this
const paymentTracker = restate.object({
    name: "PaymentTracker",
    handlers: {
        onPaymentFailed: async (ctx: restate.ObjectContext, event: StripeEvent) => {},
        onPaymentSuccess: async (ctx: restate.ObjectContext, event: StripeEvent) => {}
    }
})

export const PaymentTracker: typeof paymentTracker = {name: "PaymentTracker"};

File: go/patterns-use-cases/src/webhookcallbacks/stubs.go
package webhookcallbacks

import restate "github.com/restatedev/sdk-go"

type StripeEvent struct {
	ID      string `json:"id"`
	Type    string `json:"type"`
	Created int64  `json:"created"`
	Data    struct {
		Object struct {
			ID       string `json:"id"`
			Customer string `json:"customer"`
		} `json:"object"`
	} `json:"data"`
}

// Have a look at the scheduling tasks example (../schedulingtasks/paymentreminders.go)
// to see a full implementation of this

type PaymentTracker struct{}

func (PaymentTracker) OnPaymentFailed(ctx restate.Context, event StripeEvent) error  { return nil }
func (PaymentTracker) OnPaymentSuccess(ctx restate.Context, event StripeEvent) error { return nil }
